docs/env.md

```diff
@@ -1,51 +1,147 @@
-# Environment Variables
+## Passing Environment Variables to your application
 
-Sometimes it is practical to have different config values according to the environment that the application is running in.
+You application can run in different modes, as documented in the chapter about [Commands](commands.md) - `development`, `test`, `production`. Each of these modes might require different default values for things like:
 
-As an example:
+* log levels
+* API URLs
+* API access keys
+* ...?
 
-```js
-// config/prod.env.js
+You can define these values in the files with the name matching the mode.
+
+```javascript
+// config/variables/development.js
```

Comment:
The naming of the `variables` dir here is also a little confusing. Aren't all your env vars and webpack config settings "variables"? Maybe `config/exposed` or `config/public` would be more clear that these are the variables that will be accessible to the build/browser.

Similarly, when I started using this template, I found the fact that `config/index.js` was mostly build-related configuration a bit confusing. Maybe rename to `config/build.js` or `config/webpack`? The more explicit  import would likely make it more clear. It could also be worth a quick review of what config lives in there vs what lives in the build directory in the various webpack config files.

Perhaps `config/index` would be the file that loads various files and exposes the formats needed in different places throughout.